src/interpolations.jl

```diff
@@ -1147,3 +1176,46 @@ function value(ip::CrouzeixRaviart{2,1}, i::Int, ξ::Vec{2})
     i == 3 && return 1.0 - 2*ξ_y
     throw(ArgumentError("no shape function $i for interpolation $ip"))
 end
+
+##################################################
+# VectorizedInterpolation{<:ScalarInterpolation} #
+##################################################
+
+struct VectorizedInterpolation{vdim, refdim, refshape, order, SI <: ScalarInterpolation{refdim, refshape, order}} <: VectorInterpolation{vdim, refdim, refshape,order}
+    ip::SI
+    function VectorizedInterpolation{vdim}(ip::SI) where {vdim, refdim, refshape, order, SI <: ScalarInterpolation{refdim, refshape, order}}
+        if vdim != refdim
+            # For now this is required, since e.g.
+            #     Tensors.gradient(y -> Ferrite.value(ip, i, y), ξ)
+            # errors when extracting the gradient etc.
+            # error("embedded elements not supported yet, see https://github.com/Ferrite-FEM/Ferrite.jl/pull/651")
+        end
+        return new{vdim, refdim, refshape, order, SI}(ip)
+    end
+end
+
+# Vectorize to reference dimension by default
+function VectorizedInterpolation(ip::ScalarInterpolation{refdim}) where refdim
+    return VectorizedInterpolation{refdim}(ip)
+end
+
+Base.:(^)(ip::ScalarInterpolation, vdim::Int) = VectorizedInterpolation{vdim}(ip)
+function Base.literal_pow(::typeof(^), ip::ScalarInterpolation, ::Val{vdim}) where vdim
+    return VectorizedInterpolation{vdim}(ip)
+end
+
+# Helper to get number of copies for DoF distribution
+get_n_copies(::VectorizedInterpolation{vdim}) where vdim = vdim
+
+function getnbasefunctions(ipv::VectorizedInterpolation{vdim}) where vdim
+    return vdim * getnbasefunctions(ipv.ip)
+end
```

Comment:
Do we have some internal invariants relying on `getnbasefunctions` in combination with the new dof api?